frontends/web/src/api/lightning.ts

```diff
@@ -888,18 +887,18 @@ export const getNodeInfo = async (): Promise<NodeState> => {
 };
 
 export const getListPayments = async (params: ListPaymentsRequest): Promise<Payment[]> => {
-  return getApiResponse<Payment[]>(`lightning/list-payments?${qs.stringify(params, { skipNull: true })}`, 'Error calling getListPayments');
+  return getApiResponse<Payment[]>(`lightning/list-payments?${generateQueryString(params)}`, 'Error calling getListPayments');
```

Comment:
couldn't we just hardcoded the s param here somehow maybe:

`lightning/list-payments${params.s ? '?s=' + params.s : ''}`

Reply:
alternative: use native URLSearchParams ?
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
> couldn't we just hardcoded the s param here somehow maybe:
> 
> 
> 
> `lightning/list-payments${params.s ? '?s=' + params.s : ''}`

We could... but IMO that'd make the code less readable, as multiple if not all params are optional (there'll be a lot of ternaries). I prefer to have a utility function/service for this (more testable, so hopefully less bugs 🤞).

And yes, we could use the native URLSearchParams. I think this could simplify the function.

Reply:
I prefer the reusable function, like @shonsirsha says, the params are optional and the function is used in several places

Reply:
Ok can we try with URLSearchParams ?

It would be nice to have all optional params typed in typescript in the api/types files. When I looked at it I only saw s param.

Reply:
These are the types that need passing as query params (there will be more in the future also):

https://github.com/dangeross/bitbox-wallet-app/blob/5749d8cc2d10fb78982d95af48cdee06a68c779f/frontends/web/src/api/lightning.ts#L146-L153

https://github.com/dangeross/bitbox-wallet-app/blob/5749d8cc2d10fb78982d95af48cdee06a68c779f/frontends/web/src/api/lightning.ts#L283-L286

https://github.com/dangeross/bitbox-wallet-app/blob/5749d8cc2d10fb78982d95af48cdee06a68c779f/frontends/web/src/api/lightning.ts#L815-L817